packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider({
+            syncSummarizer: true,
+        });
+        overrideLogger = new IgnoreErrorLogger(provider.logger);
+        provider.setOverrideLogger(overrideLogger);
+
+        // Create a Container for the first client.
+        mainContainer = await createContainer();
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        connectedContainers = [];
+        closedContainers = [];
+        testNodes = new Map();
+        testNodes.set(mainDataStore.id, new TestNode(mainDataStore.id));
+        unrefDataStores = new Set();
+        refDataStores = new Set();
+        rootDataStores = new Set();
+        refDataStores.add(mainDataStore.id);
+        rootDataStores.add(mainDataStore.id);
+
+        await provider.ensureSynchronized();
+    });
+
+    afterEach(() => {
+
+    });
+
+    itExpects.skip("GC is notified when datastores are aliased.", [
+        {
+            eventName: "fluid:telemetry:Container:ContainerClose",
+            errorType: ContainerErrorType.clientSessionExpiredError,
+        },
+    ], async () => {
+        const mainContainerDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        mainContainerDataStore._root.set("test", "123");
+        mainContainerDataStore._root.set("123", "123");
+        connectedContainers.push(mainContainer);
+        await provider.ensureSynchronized();
+        const summarizeResults = mainContainerDataStore.containerRuntime.summarizeOnDemand({
+            reason: "GC Testing",
+        });
+
+        const summarySubmitted = await summarizeResults.summarySubmitted;
+        assert(summarySubmitted.success === true, "Expected successful summarizeOnDemand!");
+        assert(summarySubmitted.data !== undefined, "Expected summarizeOnDemand data!");
+        assert(summarySubmitted.data.stage === "submit", "Should have been submitted");
+        assert(summarySubmitted.data.summaryTree !== undefined, "summary tree should exist");
+        const summaryTree = summarySubmitted.data.summaryTree;
+        uploadedSummaries.push(summaryTree);
+
+        await sleep(defaultSessionExpiryDurationMs);
+
+        await loadContainer();
+        assert(mainContainer.closed === true, "Expected mainContainer to be closed!");
+        assert(uploadedSummaries.length > 0, `${uploadedSummaries.length}`);
+    });
+
+    const randomInteger = (min: number, max: number): number => {
+        return makeRandom().integer(min, max);
+    };
+
+    const loadNewContainer = async () => {
+        const newContainer = await loadContainer();
+        newContainer.on("closed", () => {
+            const index = connectedContainers.indexOf(newContainer);
+            assert(index >= 0, "Expected container to have been added to connectedContainers");
+            closedContainers.push(connectedContainers[index]);
```

Comment:
subjective nit: You can combine these two lines:  closedContainers.push(connectedContainers.splice(index, 1));

Reply:
I like the assert, so I'll just keep it in.